napari/layers/utils/color_manager.py

```diff
@@ -0,0 +1,521 @@
+        Parameters
+        ----------
+        color : Optional[ColorType]
+            The color to add. If set to None, the value of self.current_color will be used.
+            The default value is None.
+        n_colors : int
+            The number of colors to add. The default value is 1.
+        update_clims : bool
+            If in colormap mode, update the contrast limits when adding the new values
+            (i.e., reset the range to 0-new_max_value).
+        """
+        if self.mode == ColorMode.DIRECT:
+            if color is None:
+                new_color = self.current_color
+            else:
+                new_color = color
+            transformed_color = transform_color_with_defaults(
+                num_entries=n_colors,
+                colors=new_color,
+                elem_name="color",
+                default="white",
+            )
+            broadcasted_colors = normalize_and_broadcast_colors(
+                n_colors, transformed_color
+            )
+            self.colors = np.concatenate((self.colors, broadcasted_colors))
+        else:
+            # add the new value color_properties
+            color_property_name = self.color_properties.name
+            current_value = self.color_properties.current_value
+            if color is None:
+                color = current_value
+            new_color_property_values = np.concatenate(
+                (self.color_properties.values, np.repeat(color, n_colors)),
+                axis=0,
+            )
+            self.color_properties = ColorProperties(
+                name=color_property_name,
+                values=new_color_property_values,
+                current_value=current_value,
+            )
+
+            if update_clims and self.mode == ColorMode.COLORMAP:
+                self.contrast_limits = None
+
+    def remove(self, indices_to_remove: Union[set, list, np.ndarray]):
+        """Remove the indicated color elements
+        Parameters
+        ----------
+        indices_to_remove : set, list, np.ndarray
+            The indices of the text elements to remove.
+        """
+        selected_indices = list(indices_to_remove)
+        if len(selected_indices) > 0:
+            if self.mode == ColorMode.DIRECT:
+                self.colors = np.delete(self.colors, selected_indices, axis=0)
+            else:
+                # remove the color_properties
+                color_property_name = self.color_properties.name
+                current_value = self.color_properties.current_value
+                new_color_property_values = np.delete(
+                    self.color_properties.values, selected_indices
+                )
+                self.color_properties = ColorProperties(
+                    name=color_property_name,
+                    values=new_color_property_values,
+                    current_value=current_value,
+                )
+
+    def paste(self, colors: np.ndarray, properties: Dict[str, np.ndarray]):
```

Comment:
Maybe make private if generally called by layer and we don't expect users to call it?

Reply:
Done!